packages/flutter/lib/src/material/shaders/stretch_overscroll.frag

```diff
@@ -0,0 +1,157 @@
+  float distance_stretched,
+  float interpolation_strength,
+  float viewport_dimension
+) {
+  float offset_pos = in_pos - reverse_stretch_dist;
+  float pos_based_variation = mix(
+    1.0,
+    ease_in(offset_pos, u_inverse_stretch_affected_dist),
+    interpolation_strength
+  );
+  float stretch_intensity = (-overscroll) * pos_based_variation;
+  return viewport_dimension - (distance_stretched - (offset_pos / (1.0 + stretch_intensity)));
+}
+
+float compute_streched_effect(
+  float in_pos,
+  float overscroll,
+  float u_stretch_affected_dist,
+  float u_inverse_stretch_affected_dist,
+  float distance_stretched,
+  float distance_diff,
+  float interpolation_strength,
+  float viewport_dimension
+) {
+  if (overscroll > 0.0) {
+    if (in_pos <= u_stretch_affected_dist) {
+      return compute_overscroll_start(
+        in_pos, overscroll, u_stretch_affected_dist,
+        u_inverse_stretch_affected_dist, distance_stretched,
+        interpolation_strength
+      );
+    } else {
+      return distance_diff + in_pos;
+    }
+  } else if (overscroll < 0.0) {
+    float stretch_affected_dist_calc = viewport_dimension - u_stretch_affected_dist;
+    if (in_pos >= stretch_affected_dist_calc) {
+      return compute_overscroll_end(
+        in_pos,
+        overscroll,
+        stretch_affected_dist_calc,
+        u_stretch_affected_dist,
+        u_inverse_stretch_affected_dist,
+        distance_stretched,
+        interpolation_strength,
+        viewport_dimension
+      );
+    } else {
+      return -distance_diff + in_pos;
+    }
+  } else {
+    return in_pos;
+  }
+}
+
+out vec4 frag_color;
+
+void main() {
+  vec2 uv = FlutterFragCoord().xy / u_size;
+  float in_u_norm = uv.x;
+  float in_v_norm = uv.y;
+
+  float out_u_norm;
+  float out_v_norm;
+
+  float norm_stretch_affected_dist_x = 1.0;
+  float norm_stretch_affected_dist_y = 1.0;
+
+  float norm_inverse_stretch_affected_dist_x = 1.0;
+  float norm_inverse_stretch_affected_dist_y = 1.0;
+
+  float norm_distance_stretched_x = 1.0 / (1.0 + abs(u_overscroll_x));
+  float norm_distance_stretched_y = 1.0 / (1.0 + abs(u_overscroll_y));
+
+  float norm_dist_diff_x = norm_distance_stretched_x - 1.0;
+  float norm_dist_diff_y = norm_distance_stretched_y - 1.0;
+
+  float norm_viewport_width = 1.0;
+  float norm_viewport_height = 1.0;
+
+  out_u_norm = compute_streched_effect(
+    in_u_norm,
+    u_overscroll_x,
+    norm_stretch_affected_dist_x,
+    norm_inverse_stretch_affected_dist_x,
+    norm_distance_stretched_x,
+    norm_dist_diff_x,
+    u_interpolation_strength,
+    norm_viewport_width
+  );
+
+  out_v_norm = compute_streched_effect(
+    in_v_norm,
+    u_overscroll_y,
+    norm_stretch_affected_dist_y,
+    norm_inverse_stretch_affected_dist_y,
+    norm_distance_stretched_y,
+    norm_dist_diff_y,
+    u_interpolation_strength,
+    norm_viewport_height
+  );
+
+  uv.x = out_u_norm;
+  uv.y = out_v_norm;
+
+  frag_color = texture(u_texture, uv);
```

Comment:
You need to account for the reversed y axis for GLES (this is my mistake for forgetting to document).

To compute the actual uv.y you need to do something like:
```
#ifdef IMPELLER_TARGET_OPENGLES
uv.y = 1.0 - out_v_norm;
#else
uv.y = out_v_norm
#endif 
```

You can test this on an emulator which will probably use the GLES backend.